test/framework/src/main/java/org/opensearch/test/rest/yaml/section/Assertion.java

```diff
@@ -79,6 +81,41 @@ protected final Object getActualValue(ClientYamlTestExecutionContext executionCo
         return executionContext.response(field);
     }
 
+    static Object convertActualValue(Object actualValue, Object expectedValue) {
```

Comment:
@msfroh could you please explain why do we need this? I remember you've mentioned flipping between `float` and `double`, but I am curious how come we run into that now and not before?

Reply:
From what I found, the existing uses of `gt`, `lt`, `gte`, and `lte` were comparing integers (no decimal in the expected value) or they were looking at `metric_score` from the `rank-eval` plug-in, which has type `double`.

I think this is the first time that someone has tried using one of those comparisons on a `float`.

What I'm still confused about is that the test would sometimes pass. I would have expected `0.0` to consistently get parsed as `double` and the test would fail.

Reply:
Interesting, I will try to look (but not blocking the fix), thanks for explaining!